### DISCUSSION TITLE
`battery_notes_battery_not_reported` Not firing?

### DISCUSSION THREAD

**Discussion by oblivioncth** [andrew-codechimp/HA-Battery-Notes#2223] [Q&A] [answered]
Hi, awesome integration!

I'm trying to test to make sure my Battery Notes related notification automations are working, and while the low battery notification works fine, I'm having issues with the "battery not reported" condition".

I have a device that I already know is dead, and has been since before I added the integration, that I wanted to test with. It's my understanding from #2193 and:
https://github.com/andrew-codechimp/HA-Battery-Notes/blob/462d46a66fc43a6cf11b68406a49a9f27589c827/custom_components/battery_notes/__init__.py#L371

that a device with a last reported date of "Unknown" will never be considered "out of date". So, I used Developer tools to force an older date onto the device state, with the result being:

```
state_class: measurement
battery_quantity: 1
battery_type: CR2450
battery_type_and_quantity: CR2450
battery_low: false
battery_low_threshold: 15
battery_last_reported: 2024-10-10T17:02:49.605Z # I imagine this should be sufficient
battery_last_reported_level: 80 # I also bodged this
battery_last_replaced: null
device_id: 765b2201ad7f2fbf3e192baafbb8cf81
source_entity_id: 
device_name: Multisensor] Front Door Backup
unit_of_measurement: %
device_class: battery
friendly_name: Multisensor] Front Door Backup - Battery+
```

I then ran:

```
action: battery_notes.check_battery_last_reported
data:
  days_last_reported: 2
```

...and my automation never triggered:

```
triggers:
  - event_type: battery_notes_battery_not_reported
    trigger: event
```

To try and take as much out of the picture as possible, I enabled debug logging and ran the action/service again. I do see:
```
2024-10-15 13:10:48.241 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=battery_notes, service=check_battery_last_reported, service_data=days_last_reported=2>
2024-10-15 13:10:48.242 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140701488575776] Christian Heimlich from REDACTED (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36): Sending b'{"id":49,"type":"result","success":true,"result":{"context":{"id":"01JA8JJBNGAHKJ7F9A8CHAXD8S","parent_id":null,"user_id":"REDACTED"},"response":null}}'
2024-10-15 13:10:48.242 DEBUG (Recorder) [homeassistant.components.recorder.core] Processing task: <Event call_service[L]: domain=battery_notes, service=check_battery_last_reported, service_data=days_last_reported=2>
```

but never the expected log entry for when the `battery_notes_battery_not_reported` event is fired:

https://github.com/andrew-codechimp/HA-Battery-Notes/blob/462d46a66fc43a6cf11b68406a49a9f27589c827/custom_components/battery_notes/__init__.py#L394-L398

Am I doing something wrong?

The only thing I can think of is that the integration somehow stores a copy of the device state and only updates it so often, ignoring the override of the date I'm setting via Developer tools, and that the issue is indeed that "Unknown" will never trigger these events.

**Comment by andrew-codechimp** [comment]
You are correct in that the integration stores a copy of some of the device state, and since your device last seen is unknown at the start it won't trigger the event.

The data is stored in your config/.storage/battery_notes.storage file

This is an interesting problem, I wonder if it should default a last seen to the day you add it so you get these events raised, whilst  my issue #2193 is for the date last replaced defaulting it is in the same area so I'll think and investigate both together.

**Reply by oblivioncth** [reply]
Woops, I did overlook the fact that the original issue said "replaced" and not "seen", but nonetheless thanks for responding and noting the issue. At the end of the day it's a very minor problem as it doesn't come into play most of the time and will be resolved the moment I replace the batteries on this device for the first time (since installing the integration); but, I imagine it's still something worth addressing for the sake of polish/perfectionism :)

**Reply by andrew-codechimp** [reply]
No problem and thanks for raising this, I've never encountered it before but it's confusing so needs addressing.
The replaced is dependent on Battery Notes only supporting back to HA 2024.9 and I try to keep a few months behind HA releases so it's on my backlog for December or early next year.

**Comment by andrew-codechimp** [comment, marked as answer]
Version 2.3 adds support for this so closing this as complete.
